workshops/rock_paper_scissors/README.md

```diff
@@ -0,0 +1,294 @@
+
+You should see something like the following:
+
+![main.py file in Repl.it](https://cloud-lukrtynqs.vercel.app/repl.png)
+
+## Part 3: Making The Game
+![GIF](https://cloud-931ehqeec.vercel.app/coding.gif)
+
+### 1) Importing Modules
+First, we'll need to import the `randint` module. To learn more about the randint package click [here][randint].
+
+[randint]: https://www.journaldev.com/36085/randint-method-in-python/
+
+On the first line of the `main.py` file, type `from random import randint`. This will import the randint module.
+
+### 2) Giving values to the variables
+Let's make a list for all the entries a player could enter:
+```py
+t = ["r","p","s"]
+```
+'r' stands for `Rock`,'p' stands for `Paper` and 's' stands for `Scissors`.
+
+Now, let's write some code that chooses rock, paper or scissors randomly.
+```py
+computer = t[randint(0,2)]
+```
+Now we will set a variable to false and by using it in a `while` loop, we can keep the game playing infinitely! To do that, just type 
+```py
+player = False
+```
+
+Now, let's write some code so the player can enter their name.
+```py
+name = input("Enter your name:")
+```
+
+Also to make a scoring system, we'll write the following code:
+```py
+You = 0
+PC = 0 
+```
+
+Next, we want to give an option to the player to reset the game.
+```py
+print("Type 'reset' to reset score")
+```
+
+Here's the code so far:
+```py
+from random import randint
+
+t = ["r", "p", "s"]
+
+computer = t[randint(0,2)]
+
+player = False
+
+name = input("Enter your name:")
+print("Type 'reset' to reset score")
+
+You = 0
+PC = 0
+```
+
+### 3)Keeping the game in a loop
+![loop](https://cloud-nmra250be.vercel.app/loop.gif)
+
+For this, I'll use a `while` loop but you can also use `for` loops!
+
+So we'll just type:
+```py
+while Player == False: 
+```
+Now we let the player choose either `Rock` `Paper` or `Scissors`
+```
+player = input("Rock, Paper, Scissors?(r,p,s)")
+```
+
+There are 5 possibilities:
+1. The player chooses `Rock`
+2. The player chooses `Paper`
+3. The player chooses `Scissors`
+4. The word entered is invalid
+5. The player enters `reset`
+
+### 4) Producing separate outcomes for all possibilities
+
+To make things easier I will be making functions to print if the player wins or loses.
+First I will write `message = ""` This will help us by printing different message by calling the same function.
+
+Now I will be making a function that will be called only when the player wins. To do that enter the following code:
+```py
+def win():
+  global You
+  You+=1
+  print (message)
+  print('Computer =', PC, '\n', name, '=', You)
+```
+Now we will make a function that will be called only when the player will lose.
+```py
+def lose():
+  global PC
+  PC+=1
+  print (message)
+  print('Computer =', PC, '\n', name, '=', You)
+```
+Now we have to call these functions at the right places.
+
+To do that enter the following code:
+```py
```

Comment:
This code block is a little much. People will most likely copy/paste without thinking about it. I recommend breaking it down a little bit. Personally what I would do here: explain the code from the beginning of the while loop to `print(name, ‘=‘, You)`. Then, transition with something like “Next, we want to write some code for each action that someone can take.” Then give the code for “r”, and then say “what would the code be for paper and scissors?” and reveal that code block. Then explain the reset action and the `else`.

This way, This big code block is broken down a little bit and it feels more like you’re being walked through it rather than having it dumped on you.